src/coreclr/vm/jitinterface.cpp

```diff
@@ -9183,10 +9179,52 @@ CorInfoType CEEInfo::getFieldTypeInternal (CORINFO_FIELD_HANDLE fieldHnd,
 
         SigPointer ptr(sig, sigCount);
 
+        // Actual field's owner
+        MethodTable* actualFieldsOwner = field->GetApproxEnclosingMethodTable();
+
+        // Potentially, a more specific field's owner (hint)
+        TypeHandle hintedFieldOwner = TypeHandle(owner);
+
+        // Validate the hint:
+        if (!hintedFieldOwner.IsNull())
+        {
+            bool isValidParent = false;
+            if (!hintedFieldOwner.IsTypeDesc())
+            {
+                // Now we need to check whether hinted ownerTh is an actual
+                // subclass of fieldsOwner.
+
+                MethodTable* hintedFieldOwnerParent = hintedFieldOwner.AsMethodTable();
+                while (hintedFieldOwnerParent != nullptr)
+                {
+                    if (hintedFieldOwnerParent->HasSameTypeDefAs(actualFieldsOwner))
```

Comment:
You can call `MethodTable::GetMethodTableMatchingParentClass` instead (it does the same loop).

Reply:
Thanks! Switched to it